### PR TITLE
Leshan FOTA script updates

### PR DESCRIPTION
**Pull request by mike-scott**

### CODE REVIEW

**Comment by mike-scott** [comment]
Hrm, patch names for 1st and 2nd to last were incorrectly kept the same when I split them up.

File: leshan.py [hunk 1]
```diff
@@ -105,13 +105,20 @@ def run(client, url, hostname, port, monitor):
         response = get(client_list_url, raw=True)
         if response:
             for target in response:
+		Update = True
```
Comment (EmbeddedAndroid) [inline leshan.py]: PEP8 complains about these being upper case... "Use the function and variable naming rules: lowercase with words separated by underscores as necessary to improve readability."

File: leshan.py [hunk 2]
```diff
@@ -105,13 +105,20 @@ def run(client, url, hostname, port, monitor):
...
-                    if ret:
-                         print "%s update completed" % target['endpoint']
-                    else:
-                        print "%s failed to update, aborting..." % target['endpoint']
-                        exit(1)
+                    if device:
+                        endpoint_url = 'http://%s:%s/api/clients/%s/3/0/1'  % (hostname, port, target['endpoint'])
+			endpoint_device = get(endpoint_url)
+                        if (endpoint_device != device):
+                            Update = False
```
Comment (EmbeddedAndroid) [inline leshan.py]: ditto here.

File: leshan.py [hunk 3]
```diff
@@ -105,13 +105,20 @@ def run(client, url, hostname, port, monitor):
...
+                    if Update:
```
Comment (EmbeddedAndroid) [inline leshan.py]: and here too.

File: leshan.py [hunk 4]
```diff
@@ -134,9 +221,10 @@ def main():
     parser.add_argument('-host', '--hostname', help='Leshan server hostname or ip', default='leshan')
     parser.add_argument('-port', '--port', help='Leshan server port', default='8080')
     parser.add_argument('-m', '--monitor', help='Monitor the update', action='store_true', default=False)
-    parser.add_argument('-d', '--device', help='Device tyepe filter', default=None)
+    parser.add_argument('-d', '--device', help='Device type filter', default=None)
+    parser.add_argument('-t', '--threads', help='Maximum threads', default=1)
```
Comment (EmbeddedAndroid) [inline leshan.py]: You can add type=int here to type check the inputs, instead of recasting the input below.

**Review by EmbeddedAndroid — CHANGES_REQUESTED** [review]
Couple of nits, otherwise looks like a good series. Thanks Mike!

**Comment by mike-scott** [comment]
Changed the "Update" var to "perform_update".  Diff: https://hastebin.com/tuqihoduju.sql

**Review by EmbeddedAndroid — APPROVED** [review]

**Pull request merged**